pkg/manager/member/pd_member_manager.go

```diff
@@ -454,13 +454,15 @@ func (m *pdMemberManager) getNewPDServiceForTidbCluster(tc *v1alpha1.TidbCluster
 			Selector: pdSelector.Labels(),
 		},
 	}
-	// if set pd service type ,overwrite global variable services
+
+	// override fields with user-defined ServiceSpec
 	svcSpec := tc.Spec.PD.Service
 	if svcSpec != nil {
 		if svcSpec.Type != "" {
 			pdService.Spec.Type = svcSpec.Type
 		}
-		pdService.ObjectMeta.Annotations = CopyAnnotations(svcSpec.Annotations)
+		pdService.ObjectMeta.Annotations = util.CopyStringMap(svcSpec.Annotations)
```

Comment:
do we need to _combine_ annotations?

Reply:
`pdService.ObjectMeta.Annotations` is always nil here